tests/Tokenizer/TokensAnalyzerTest.php

```diff
@@ -499,34 +499,54 @@ public function testIsAnonymousClass(string $source, array $expected): void
         }
     }
 
-    public function provideIsAnonymousClassCases(): array
+    public function provideIsAnonymousClassCases(): \Generator
     {
-        return [
-            [
-                '<?php class foo {}',
-                [1 => false],
-            ],
-            [
-                '<?php $foo = new class() {};',
-                [7 => true],
-            ],
-            [
-                '<?php $foo = new class() extends Foo implements Bar, Baz {};',
-                [7 => true],
-            ],
-            [
-                '<?php class Foo { function bar() { return new class() {}; } }',
-                [1 => false, 19 => true],
-            ],
-            [
-                '<?php $a = new class(new class($d->a) implements B{}) extends C{};',
-                [7 => true, 11 => true],
-            ],
-            [
-                '<?php interface foo {}',
-                [1 => false],
-            ],
+        yield [
+            '<?php class foo {}',
+            [1 => false],
+        ];
+
+        yield [
+            '<?php $foo = new class() {};',
+            [7 => true],
         ];
+
+        yield [
+            '<?php $foo = new class() extends Foo implements Bar, Baz {};',
+            [7 => true],
+        ];
+
+        yield [
+            '<?php class Foo { function bar() { return new class() {}; } }',
+            [1 => false, 19 => true],
+        ];
+
+        yield [
+            '<?php $a = new class(new class($d->a) implements B{}) extends C{};',
+            [7 => true, 11 => true],
+        ];
+
+        yield [
+            '<?php interface foo {}',
+            [1 => false],
+        ];
+
+        yield [
+            '<?php $foo = new /* */ class() {};',
+            [9 => true],
+        ];
+
+        if (\PHP_VERSION_ID >= 80000) {
```

Comment:
i'm not fully happy myself with conditional cases that are not displayed in PHPUnit outcome as "skipped"

Reply:
Can you elaborate why? We decided [once](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5423) to disable verbose output of PHPUnit tests because of too many skipped tests, but verbose output might be useful sometimes.

The reason I can think of is that is easier to make an unnoticed type like `800000` that a typo in `@requires 8.0`. On the other side making `testFix`, `testFix74`, `testFix80` (and soon `81`) where the only difference is an `@requires` annotation is a waste.

Maybe a feature that allow mixing `@dataProvider` and `@requires` annotation would solve that? @sebastianbergmann are you planning something like this or would consider if implementation was made?

Reply:
because of false feeling that "everything is perfect" when I have `100 / 100` tests passing while same time 50 of them were not executed due to requirements. I prefer to see `100 / 150 green, 50 skipped` in such case.

Reply:
> Maybe a feature that allow mixing `@dataProvider` and `@requires` annotation would solve that? @sebastianbergmann are you planning something like this or would consider if implementation was made?

I do not understand what you are proposing, sorry. Please open a ticket for PHPUnit with a simple example of what you propose. Thanks!